### PR TITLE
Implement `CurrencyDisplaynameV1`

### PR DESCRIPTION
**Pull request by younies**

### CODE REVIEW

File: provider/source/src/currency/displayname.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            metadata: Default::default(),
+            payload: DataPayload::from_owned(CurrencyDisplaynameV1 {
+                display_name: Cow::Owned(
+                    currency
+                        .display_name
+                        .as_deref()
+                        .or(currency.one.as_deref())
+                        .ok_or(DataError::custom(
+                            "No display name or `one` count found for the currency",
+                        ))?
```
Comment (robertbastian) [inline provider/source/src/currency/displayname.rs:42]: This should be `DataErrorKind::IdentifierNotFound`.

Comment (younies) [reply provider/source/src/currency/displayname.rs]: done

File: components/experimental/src/dimension/provider/currency_displayname.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::dimension::provider::currency_displayname)
+)]
+#[yoke(prove_covariance_manually)]
+pub struct CurrencyDisplaynameV1<'data> {
+    /// The display name for the currency.
+    /// NOTE:
+    ///     If the display name is not found, the `one` count will be used.
```
[the remark this answers is not in the document]
Comment (younies) [reply components/experimental/src/dimension/provider/currency_displayname.rs]: done

File: provider/source/src/currency/displayname.rs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                .read_and_parse(&locale, "currencies.json")?;
+
+            let currencies = &currencies_resource.main.value.numbers.currencies;
+            for (currency, patterns) in currencies {
+                // If the display name is not found, the `one` count will be used.
+                // Therefore, we can skip any currencies that don't have a display name or `one` count.
+                if patterns
+                    .display_name
+                    .as_ref()
+                    .or(patterns.one.as_ref())
```
[the remark this answers is not in the document]
Comment (younies) [reply provider/source/src/currency/displayname.rs]: done

File: provider/source/src/currency/displayname.rs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            .ok_or(DataError::custom("No currency associated with the aux key"))?;
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: DataPayload::from_owned(CurrencyDisplaynameV1 {
+                display_name: Cow::Owned(
+                    currency
+                        .display_name
+                        .as_deref()
+                        .or(currency.one.as_deref())
```
[the remark this answers is not in the document]
Comment (younies) [reply provider/source/src/currency/displayname.rs]: done

File: provider/source/src/currency/displayname.rs [hunk 5]
```diff
@@ -36,10 +40,7 @@ impl DataProvider<CurrencyDisplaynameV1Marker> for crate::SourceDataProvider {
                     currency
                         .display_name
                         .as_deref()
-                        .or(currency.one.as_deref())
-                        .ok_or(DataError::custom(
-                            "No display name or `one` count found for the currency",
-                        ))?
+                        .ok_or(DataError::custom("No display name found for the currency"))?
```
Comment (robertbastian) [inline provider/source/src/currency/displayname.rs:43]: this should also be `IdentifierNotFound`

**Review by robertbastian — COMMENTED** [review]
just one nit

File: provider/source/src/currency/displayname.rs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        let currency = currencies_resource
+            .main
+            .value
+            .numbers
+            .currencies
+            .get(req.id.marker_attributes.as_str())
+            .ok_or_else(|| {
+                DataErrorKind::IdentifierNotFound
+                    .into_error()
+                    .with_debug_context("No currency associated with the aux key")
```
Comment (robertbastian) [inline provider/source/src/currency/displayname.rs:33]:
```suggestion
                    .with_debug_context("No data for currency")
```

File: provider/source/src/currency/displayname.rs (hunk 5 above)
Comment (younies) [reply provider/source/src/currency/displayname.rs]: done

**Review by robertbastian — APPROVED** [review]

**Review by robertbastian — APPROVED** [review]

**Pull request merged**